### PR TITLE
Add waiter for ssh to nodes

### PR DESCRIPTION
**Pull request by bkopilov**
When starting a node and we need access to the node it may take some time till node will be ready and sshable from outside.
ping or verify ip is not enough in cases ssh service not ready.

Added a waiter in nodes.py

### CODE REVIEW

**Comment by bkopilov** [comment]
@lalon4 , @osherdp

**Review by osherdp — APPROVED** [review]

**Comment by osherdp** [comment]
/hold
lint fails

**Comment by bkopilov** [comment]
Yes i see my fault

/src/assisted_test_infra/test_infra/helper_classes/nodes.py:237:25: B023 Function definition does not bind loop variable 'node'.  intersting ....

File: src/assisted_test_infra/test_infra/helper_classes/nodes.py [hunk 1]
```diff
@@ -228,3 +229,24 @@ def _are_nodes_network_prepared(self):
...
+            except (TimeoutError, SCPException):
+                return False
+            else:
+                return True
+
+        waiting.wait(
+            lambda: _ssh_connection(),
+            timeout_seconds=180,
+            sleep_seconds=20,
+            waiting_for="Nodes to allow ssh connection",
```
Comment (osherdp) [inline src/assisted_test_infra/test_infra/helper_classes/nodes.py:250]: few fixes to make the code more clear:
```suggestion
        def _all_nodes_allow_ssh_connection():
            try:
                for node in self.nodes:
                    if node.ssh_connection is None:
                        return False
            except (TimeoutError, SCPException):
                return False

            return True

        waiting.wait(
            lambda: _all_nodes_allow_ssh_connection(),
            timeout_seconds=180,
            sleep_seconds=20,
            waiting_for="nodes to allow ssh connection",
```

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/assisted_test_infra/test_infra/helper_classes/nodes.py (hunk 1 above)
Comment (bkopilov) [reply src/assisted_test_infra/test_infra/helper_classes/nodes.py:250]: > few fixes to make the code more clear:

Ack no problem .

Comment (osherdp) [reply src/assisted_test_infra/test_infra/helper_classes/nodes.py]: hmm, you still didn't apply all those changes. please look carefully

**Comment by bkopilov** [comment]
https://github.com/openshift/assisted-test-infra/pull/2026/files

sorry , what did i miss ?   by the way good catch which the bug that i
created with closures that was caught :)

**Review by osherdp — APPROVED** [review]

**Pull request merged**